events/sinks/influxdb/influxdb.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package influxdb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/heapster/events/core"
+	metrics_core "k8s.io/heapster/metrics/core"
+	influxdb_common "k8s.io/heapster/util/influxdb"
+	kube_api "k8s.io/kubernetes/pkg/api"
+
+	"github.com/golang/glog"
+	influxdb "github.com/influxdb/influxdb/client"
+)
+
+type influxdbSink struct {
+	client influxdb_common.InfluxdbClient
+	sync.RWMutex
+	c        influxdb_common.InfluxdbConfig
+	dbExists bool
+}
+
+const (
+	eventMeasurementName = "log/events"
+	// Event special tags
+	eventUID = "uid"
+	// Value Field name
+	valueField = "value"
+	// Event special tags
+	dbNotFoundError = "database not found"
+
+	// Maximum number of influxdb Points to be sent in one batch.
+	maxSendBatchSize = 1000
+)
+
+func (sink *influxdbSink) resetConnection() {
+	glog.Infof("Influxdb connection reset")
+	sink.dbExists = false
+	sink.client = nil
+}
+
+// Generate point value for event
+func getEventValue(event *kube_api.Event) (string, error) {
+	bytes, err := json.MarshalIndent(event, "", " ")
```

Comment:
Why do you want to indent this? I think it will consume more bytes as a result and since you sent it to influxdb anyway it probably doesn't matter how it's formatted.